test/mocha-junit-reporter-spec.js

```diff
@@ -536,120 +617,98 @@ describe('mocha-junit-reporter', function() {
     };
 
     it('should generate valid testCase for testCaseSwitchClassnameAndName default', function() {
-      reporter = createReporter({mochaFile: 'test/mocha.xml'});
+      var reporter = createReporter();
       var testCase = reporter.getTestcaseData(mockedTestCase);
       expect(testCase.testcase[0]._attr.name).to.equal(mockedTestCase.fullTitle());
       expect(testCase.testcase[0]._attr.classname).to.equal(mockedTestCase.title);
     });
 
     it('should generate valid testCase for testCaseSwitchClassnameAndName=false', function() {
-      reporter = createReporter({mochaFile: 'test/mocha.xml', testCaseSwitchClassnameAndName: false});
+      var reporter = createReporter({testCaseSwitchClassnameAndName: false});
       var testCase = reporter.getTestcaseData(mockedTestCase);
       expect(testCase.testcase[0]._attr.name).to.equal(mockedTestCase.fullTitle());
       expect(testCase.testcase[0]._attr.classname).to.equal(mockedTestCase.title);
     });
 
     it('should generate valid testCase for testCaseSwitchClassnameAndName=true', function() {
-      reporter = createReporter({mochaFile: 'test/mocha.xml', testCaseSwitchClassnameAndName: true});
+      var reporter = createReporter({testCaseSwitchClassnameAndName: true});
       var testCase = reporter.getTestcaseData(mockedTestCase);
       expect(testCase.testcase[0]._attr.name).to.equal(mockedTestCase.title);
       expect(testCase.testcase[0]._attr.classname).to.equal(mockedTestCase.fullTitle());
     });
   });
 
   describe('XML format', function () {
-    var suites = [
-      {testsuite:
-        {title: '', root: true, suites: [2], tests: [0]}
-      },
-      {testsuite:
-        {title: 'Inner Suite', suites: [1], tests: [1]}, pass: [
-          {title: 'test', fullTitle: 'Inner Suite test'}
-        ]
-      },
-      {testsuite:
-        {title: 'Another Suite', suites: [1], tests: [1]}, fail: [
-          {title: 'fail test', fullTitle: 'Another Suite fail test', error: new Error('failed test')}
-        ]
-      }
-    ];
+    it('generates Jenkins compatible XML when in jenkinsMode', function(done) {
+      this.timeout(10000); // xmllint is very slow
```

Comment:
We may want to replace xmllint with something else. I think we were close to the 2s limit before but actually running mocha caused us to go over it more often.

Reply:
Thanks for the feedback, I actually just put `xmllint` in because our previous solution doesn't appear to support more recent node releases. I'm willing to wait a couple seconds if it means no native libraries to build and link.

Reply:
It looks like all the alternatives either use libxml or Java, so `xmllint` may be the best solution after all.